### ISSUE TITLE
page 0, till you scroll down the entirety of the book

### ISSUE THREAD

**Issue by MysterieDev** [typemill-resources/ebooks#8]
Hey, i really like the new layout "report". 

I've tried it out on my educational content.
One thing that confused me at first when generating a book was the following:
![image](https://user-images.githubusercontent.com/32373099/156715681-cebbace3-f83b-4f7c-8ae0-062eaade6511.png)
here you can see page/overallpages.

Weirdly, till you scroll down the book in its entirety,  it displays the overall pages as *0*
![image](https://user-images.githubusercontent.com/32373099/156715814-c8f0d2c1-6c99-4995-ae95-3ad9febb8bbc.png)

it refreshes itself, when i scroll down to the lastpage.

im using the newest version of typemill.

**Comment by trendschau** [comment]
I am glad that you like it and yes, that is the behavior of paged.js that is used for the ebook-preview. I have to add that behavior to the description in the next release. 

I cannot really explain but pagejs renders all the pages in the browser first and after that does the final paging. Pagedjs does that in multiple steps and scrolling down seems to be required to finish the first step. I cannot explain the details but the general flow is explained on the website of paged.js: https://pagedjs.org/documentation/4-how-paged.js-works/

**Comment by MysterieDev** [comment]
@trendschau okay, got it!

Maybe there is a possiblity to scroll programmatically down and up to the first page again?

I see that some content is - kind of - lazy loaded in upon scrolling down.

Maybe I'll look into paged.js a bit.

**in general**
I think the ebook plugin is a feature that actually is the unique selling point of typemill against GRAV, etc.. and i'd love to see it getting a bit less "clunky" in its feeling. Of course, if used right, as i can see i.E. in CMSStash, it already can create beautiful publications.

For example i see custom markdown like the TOC or YouTube Videos not being the perfect fit for a book format.

I thought about trying out a additional build step in content creation where everything, that could destroy the ebook-layout or is nonsense to put in (youtube thumbnails for example) get thrown out before the book is created from the files.

**Comment by trendschau** [comment]
Yes, you are absolutely right, and the ebook feature is one focus of further development. Unfortunately it is quite big and complex and my resources are a bit limited. 

Regarding the content: Yes, with the latest version I added the possibility to deactivate shortcode-content because this is often unwanted in ebooks. Other content like youtube (I personally use them and link in the context to youtube, in PDF reading on connected devices this can make sense), TOC and others. 

Another big topic is a print version for ebooks, which has some more requirements like recto-verso, printmarks, differently styled links and references and more. I did not had the time to work on that. The next topic is stuff like math integration which is really complicated, unfortunately. And the epub, which is pretty low level and basic right now. 

So much space for improvements but maybe making the current features less clunky from UX perspective is the best next step.
